engines/scumm/actor.cpp

```diff
@@ -2489,11 +2489,15 @@ void ActorHE::prepareDrawActorCostume(BaseCostumeRenderer *bcr) {
 
 	if (_vm->_game.heversion >= 80 && _heNoTalkAnimation == 0 && _animProgress == 0) {
 		if (_vm->getTalkingActor() == _number && !_vm->_string[0].no_talk_anim) {
-			int talkState = 0;
+			int talkState = -1;
 
-			if (((SoundHE *)_vm->_sound)->isSoundCodeUsed(1))
-				talkState = ((SoundHE *)_vm->_sound)->getSoundVar(1, 19);
-			if (talkState == 0)
+			if (!((SoundHE *)_vm->_sound)->isSoundCodeUsed(HSND_TALKIE_SLOT))
+				talkState = _vm->_rnd.getRandomNumberRng(1, 10);
+			else
+				talkState = ((SoundHE *)_vm->_sound)->getSoundVar(HSND_TALKIE_SLOT, 19);
+
+			// Allow a talkie with tokens to kick into random mouth mode
+			if (talkState == -1 || talkState == 0)
```

Comment:
Perhaps we can have explicit name for these states (rather than -1, 0)?
how does talkState == 0 can happen?

Reply:
These are unnamed even in the original code; they correspond to a range of faces the character can assume during lipsync.
Oh believe me, the 0 case does happen (value fetched from sound variables) and crashes the game 😄
The original code checks for 0 too but it does that in the function call which follows immediately after (their setTalkCondition()).

Reply:
I believe this could be simplified into something like this:
```
int talkState = -1;

if ((SoundHE *)_vm->_sound)->isSoundCodeUsed(HSND_TALKIE_SLOT)
    talkState = ((SoundHE *)_vm->_sound)->getSoundVar(HSND_TALKIE_SLOT, 19);

// Allow a talkie with tokens to kick into random mouth mode
if (talkState == -1 || talkState == 0)
    talkState = _vm->_rnd.getRandomNumberRng(1, 10);
```

Reply:
Thanks, committed! 🙏